rtCommon/imageHandling.py

```diff
@@ -274,9 +274,23 @@ def convertDicomFileToNifti(dicomFilename, niftiFilename):
     subprocess.run(cmd, shell=False, stdout=subprocess.DEVNULL)
 
 
-def readNifti(niftiFilename):
-    nifitImg = nib.load(niftiFilename)
-    return nifitImg
+def niftiToMem(niftiImg):
+    '''
+    Fully load Nifti image into memory and remove any file-backing.
+    NiftiImage by default contains a pointer to the image file for the data.
+    '''
+    niftiBytes = niftiImg.to_bytes()
+    niftiMemImg = niftiImg.__class__.from_bytes(niftiBytes)
+    return niftiMemImg
+
+
+def readNifti(niftiFilename, memCached=True):
+    niftiImg = nib.load(niftiFilename)
+    # When memCached is True we want to load the image all in memory so, for example,
+    #   it can be sent remotely.
+    if memCached is True:
+        niftiImg = niftiToMem(niftiImg)
```

Comment:
I'm wondering whether using `nib.load(..., mmap=False)` might be more efficient here than a full serialization/deserialization round-trip

Reply:
It is faster (I wrote a quick test to check), but it doesn't appear to be equivalent. The niftiImage object that results still depends on the underlying nifti file being present.